src/app/layout/Page/index.tsx

```diff
@@ -47,43 +50,67 @@ const PageContainer = ({ children, ...rest }) => {
 type PageTopBarProps = FlexProps & {
   onBack?(): void;
   showBack?: boolean;
+  isFixed?: boolean;
 };
 
 export const PageTopBar = ({
   children,
   onBack = () => undefined,
   showBack = false,
+  isFixed = false,
   ...rest
 }: PageTopBarProps) => {
+  const topBarRef = useRef<HTMLDivElement>(null);
+  const [height, setHeight] = useState(0);
+  const { isFocusMode } = useLayoutContext();
+  const theme = useTheme();
+
+  useLayoutEffect(() => {
+    const onResize = () => {
+      setHeight(topBarRef.current?.offsetHeight || 0);
+    };
+    onResize();
+    window.addEventListener('resize', onResize);
+    return () => {
+      window.removeEventListener('resize', onResize);
+    };
+  }, []);
+
   const { rtlValue } = useRtl();
   const { colorModeValue } = useDarkMode();
   return (
-    <Flex
-      zIndex="2"
-      direction="column"
-      pt="4"
-      pb="4"
-      boxShadow="0 4px 20px rgba(0, 0, 0, 0.05)"
-      bg={colorModeValue('white', 'gray.900')}
-      {...rest}
-    >
-      <Box w="full" h="0" pb="safe-top" />
-      <PageContainer>
-        <HStack spacing="4">
-          {showBack && (
-            <Box ms={{ base: 0, lg: '-3.5rem' }}>
-              <IconButton
-                aria-label="Go Back"
-                icon={rtlValue(<FiArrowLeft />, <FiArrowRight />)}
-                variant="ghost"
-                onClick={() => onBack()}
-              />
-            </Box>
-          )}
-          <Box flex="1">{children}</Box>
-        </HStack>
-      </PageContainer>
-    </Flex>
+    <>
+      {isFixed && <Box h={`${height}px`} />}
+      <Flex
+        zIndex="2"
+        direction="column"
+        pt="4"
+        pb="4"
+        boxShadow="0 4px 20px rgba(0, 0, 0, 0.05)"
+        bg={colorModeValue('white', 'gray.900')}
+        ref={topBarRef}
+        top={isFocusMode ? 0 : theme.layout.topBar.height}
```

Comment:
This should also be conditional if the PageTopBar is not fixed

Reply:
fixed